.github/workflows/build-and-test.yml

```diff
@@ -471,7 +471,7 @@ jobs:
         run: |
           cp bin/* dist/
           cd dist && shasum -a 256 * > checksums.txt && cd ../
-          ghr -t $GITHUB_TOKEN -u "GitHub Action" -r opentelemetry-collector-contrib --replace $RELEASE_TAG dist/
+          ghr -t $GITHUB_TOKEN -u open-telemetry -r opentelemetry-collector-contrib --replace $RELEASE_TAG dist/
```

Comment:
Not a big deal but noticed that perhaps in the future, this can be replaced with the official `gh` CLI which is preinstalled in actions, to reduce dependencies on third party tools. It seems to have better UX, not requiring these `-u`, `-r` parameters.

https://cli.github.com/manual/gh_release_create

```
Upload all tarballs in a directory as release assets
$ gh release create v1.2.3 ./dist/*.tgz
```

Reply:
In the long term i'd like to replace all of the steps to release/build w/ goreleaser, to be consistent across all otel collector repos.